### PR TITLE
*: put goyacc and golex tool into vendor

### PR DESCRIPTION
**Pull request by tiancaiamao**
call `make` should not go get github.com/pingcap/goyacc now
PTAL @siddontang @shenli

### CODE REVIEW

**Comment by siddontang** [comment]
do you test it?

**Comment by tiancaiamao** [comment]
move golex and goyacc main.go files directly to `parser/*` directory, and add vendor.
now qiuyesuifeng/golex and pingcap/goyacc are not needed any more...

PTAL @siddontang @shenli

**Comment by coocood** [comment]
LGTM

File: Makefile [hunk 1]
```diff
@@ -53,9 +53,15 @@ install:
 
 TEMP_FILE = temp_parser_file
 
-parser:
-	go get github.com/pingcap/goyacc
-	go get github.com/qiuyesuifeng/golex
+golex:
+	$(GO) get github.com/qiuyesuifeng/golex
```
Comment (siddontang) [inline Makefile]: still go get golex?

Comment (tiancaiamao) [inline Makefile]: we'll get rid of golex tool sooner: https://github.com/pingcap/tidb/issues/1490 ,
put golex into vendor increase lot of files, so maybe it's cleaner this way.

**Comment by shenli** [comment]
LGTM

**Pull request merged**